src/ixdat/techniques/deconvolution.py

```diff
@@ -0,0 +1,254 @@
+
+        return sig_area
+
+
+    @property
+    def charge(self):
+        """Integrates the measured current over the time of the impulse and returns
+        the charge passed.
+        """
+        y_curr = self.EC_data[1]
+
+        mask = np.isclose(
+            y_curr,
+            y_curr[0],
+            rtol=1e-1
+            )
+
+        Q = np.trapz(y_curr[mask], self.EC_data[0][mask])
+
+        return Q
+
+    def plot(self,
+        time=None,
+        ax = None,
+        norm=True,
+        **kwargs
+        ):
+        """Returns a plot of the kernel/impulse response."""
+        if ax is None:
+            fig1 = plt.figure()
+            ax = fig1.add_subplot(111)
+
+        if self.type is 'functional':
+            ax.plot(
+                time, self.gen_kernel(time=time, norm=norm),
+                **kwargs,
+                )
+
+        elif self.type is 'measured':
+            ax.plot(
+            self.MS_data[0], self.gen_kernel(norm=norm),
+            **kwargs,
+            )
+
+        else:
+            raise Exception('Nothing to plot with blank kernel')
+
+
+        return ax
+
+    def gen_kernel(self,
+        dt=0.1,
+        len=100,
+        norm=True,
+        matrix=False
+        ):
+        """Generates a kernel/impulse response based on the defined mass
+        transport parameters contained in dict.
+
+        Args:
+            dt (int): Timestep for which the kernel/impulse response is calculated.
+            len(int): Timelength in seconds for which the kernel/impulse response is
+                calculated.
+            norm (bool): If true the kernel/impulse response is normalized to its
+                area.
+            matrix (bool): If true the circulant matrix constructed from the kernel/
+                impulse reponse is returned.
+        """
+        if self.type is 'functional':
+
+            time = np.arange(0,len,dt)
+            time[0] = 1e-6
+
+            D = self.params['D']
+            L = self.params['L']
+            V = self.params['V']
+            V_dot = self.params['V_dot']
+            kH = self.params['kH']
+
+            tdiff = time * D / (L**2)
+            fs = lambda s: 1/(
+                sqrt(s) * sinh(sqrt(s))
+                + (V * kH / 0.196e-4 / L)
+                * (s + V_dot / V * L**2 / D)
+                * cosh(sqrt(s))
+                )
+
+            kernel = np.zeros(len(time))
+            for i in range(len(time)):
+                kernel[i] = invertlaplace(fs, tdiff[i], method='talbot')
+
+        elif self.type is 'measured':
+            kernel = self.MS_data[1]
+            time = self.MS_data[0]
+
+        if norm:
+            area = np.trapz(kernel, time)
+            kernel = kernel/area
+
+        if matrix:
+            kernel = np.tile(kernel, (len(kernel), 1))
+            i = 1
+            while i < len(time):
+                kernel[i] = np.concatenate((kernel[0][i:], kernel[0][:i]))
+                i=i+1
+
```

Comment:
I would cache the kernel, by including a line
`self._kernel = kernel` in this method,
and then adding a property which accesses this and calculates it if needed 
```
@property 
def kernel(self):
    """The numerical deconvolution kernel (impulse response function), vs self.t"""
    if not self._kernel:
        self.calculate_kernel()   # renamed from gen_kernel
    return self._kernel
```
Also, store the time used for the kernel (`self.t = t_kernel  # renamed from 'time'`), and this simplifies the plotting method:
```
def plot_kernel(self):
    ...
    kernel = self.kernel
    ax.plot(self.t, kernel)
    ...
```

Reply:
A more radical (but not too difficult) makeover that we should consider is for `Kernel` to inherit from `Measurement` and store its `t` and `kernel` as series - this would give it access to the backend for saving and loading. Do you think that would be useful?